mantle/platform/api/aliyun/api.go

```diff
@@ -385,3 +387,36 @@ func (a *API) ListRegions() ([]string, error) {
 	sort.Strings(ret)
 	return ret, nil
 }
+
+// MakePublic modifies an image uploaded to Aliyun as publicly available
+// NOTE: only us-east-1 and us-west-1 support making images public unless
+// your account has been allowlisted by Aliyun to operate on all regions
+func (a *API) MakePublic(id string, region string) error {
+	// Need to check if the image is already public
+	images, err := a.GetImagesByID(id, region)
+	if err != nil {
+		return fmt.Errorf("getting image: %v", err)
+	}
+	if images.TotalCount < 1 {
+		return fmt.Errorf("no image found with id %v", id)
+	}
+
+	for _, img := range images.Images.Image {
+		if img.ImageId == id && img.IsPublic != true {
+			request := ecs.CreateModifyImageSharePermissionRequest()
+			request.SetConnectTimeout(defaultConnectTimeout)
+			request.SetReadTimeout(defaultReadTimeout)
+			request.Scheme = "https"
+			request.ImageId = id
+			request.RegionId = region
+			request.IsPublic = requests.NewBoolean(true)
+
+			_, err := a.ecs.ModifyImageSharePermission(request)
+			return err
+		} else {
+			plog.Infof("image %v already public", id)
+			return nil
```

Comment:
Doesn't this short-circuit the loop?

I'm not clear TBH when `GetImagesByID` can return multiple images. Can multiple images have the same ID somehow?